src/SDK/Metrics/MetricReader/ExportingReader.php

```diff
@@ -107,6 +107,8 @@ public function forceFlush(): bool
         $collect = $this->doCollect();
         $forceFlush = $this->exporter->forceFlush();
 
+        $this->sources = [];
```

Comment:
I'm not convinced that forceFlush should clear sources, let's see if @Nevay can provide any guidance. Is there something in the spec that says that it should behave that way (or a bug that's bring exhibited that this fixes?)

Reply:
Sources must not be cleared.

Regarding the issue: `InMemoryExporter::collect()` does __not__ reset collected metrics by default, `$exporter->collect(true)` has to be used if collected metrics should be cleared.

Reply:
I used `InMemoryExporter` for simple reproduction. I have the same issue with `Otlp*Exporter`. With the current implementation, exporters just send the metrics given to them.